From b4fb1bb3edf508123ff89af357800b82072dc764 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Sat, 11 May 2019 10:23:08 +0200
Subject: [PATCH] meson: fix TLS support under mingw

GCC on Windows complains that "__declspec(thread)" doesn't work, but still
compiles it, so the meson check doesn't work.

Skip the __declspec check when gcc is used instead.

This fixes the test suite (minus gtk tests) on Windows with mingw.
---
 meson.build | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 1a11587..cb4e4c0 100644
--- a/meson.build
+++ b/meson.build
@@ -407,7 +407,8 @@ foreach h : ['sys/mman.h', 'fenv.h', 'unistd.h']
   endif
 endforeach
 
-if (host_machine.system() == 'windows' and
+# gcc on Windows only warns that __declspec(thread) isn't supported
+if (host_machine.system() == 'windows' and cc.get_id() != 'gcc' and
     cc.compiles('int __declspec(thread) foo;', name : 'TLS via __declspec(thread)'))
   config.set('TLS', '__declspec(thread)')
 elif cc.compiles('int __thread foo;', name : 'TLS via __thread')
-- 
2.21.0

